### PR TITLE
[MM-57726] Convert AddIncomingWebhook from Class Component to Function Component

### PR DESCRIPTION
**Pull request by MeHow25**
#### Summary
Convert AddIncomingWebhook from Class Component to Function Component

#### Ticket Link
Fixes https://github.com/mattermost/mattermost/issues/26694
Jira https://mattermost.atlassian.net/browse/MM-57726

#### Release Note
```
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/integrations/add_incoming_webhook/add_incoming_webhook.test.tsx [hunk 1]
```diff
@@ -26,16 +63,29 @@ describe('components/integrations/AddIncomingWebhook', () => {
     });
 
     test('should have called createIncomingHook', () => {
-        const hook = TestHelper.getIncomingWebhookMock({
-            channel_id: 'channel_id',
+        const hook = {
```
Comment (larkox) [inline webapp/channels/src/components/integrations/add_incoming_webhook/add_incoming_webhook.test.tsx:66]: Why do we remove the call to the TestHelper?

File: webapp/channels/src/components/integrations/add_incoming_webhook/add_incoming_webhook.tsx [hunk 2]
```diff
@@ -43,45 +43,38 @@ type Props = {
...
+    team,
+    enablePostUsernameOverride,
+    enablePostIconOverride,
+    actions,
+}: Props) => {
+    const [serverError, setServerError] = useState('');
 
-    addIncomingHook = async (hook: IncomingWebhook) => {
-        this.setState({serverError: ''});
+    const addIncomingHook = async (hook: IncomingWebhook) => {
```
Comment (larkox) [inline webapp/channels/src/components/integrations/add_incoming_webhook/add_incoming_webhook.tsx:54]: Let's use a `useCallback` so we don't pass a new function on every render.

**Review by larkox — CHANGES_REQUESTED** [review]
Great work! One minor change, and a question.

File: webapp/channels/src/components/integrations/add_incoming_webhook/add_incoming_webhook.tsx (hunk 2 above)
Comment (MeHow25) [reply webapp/channels/src/components/integrations/add_incoming_webhook/add_incoming_webhook.tsx]: Done

File: webapp/channels/src/components/integrations/add_incoming_webhook/add_incoming_webhook.test.tsx (hunk 1 above)
Comment (MeHow25) [reply webapp/channels/src/components/integrations/add_incoming_webhook/add_incoming_webhook.test.tsx:66]: This call was returning an object with some additional properties like `create_at` or `delete_at` that I assumed they are not relevant in this test. Do you think this call should stay?

Comment (larkox) [reply webapp/channels/src/components/integrations/add_incoming_webhook/add_incoming_webhook.test.tsx:66]: If I am not mistaken, it should stay, yes. The idea of this call is to mock the default values so it looks like the object you would naturally receive out in the wild. If those extra values are creating any issue in terms of types or anything, we can look a bit deeper why that is, but it is better if we use it.

Comment (MeHow25) [reply webapp/channels/src/components/integrations/add_incoming_webhook/add_incoming_webhook.test.tsx]: Ok, I get it :) I've added these additional properties. Now the test is comparing whole objects, and it seems to work :+1:.

**Review by larkox — APPROVED** [review]

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by yasserfaraazkhan — APPROVED** [review]
adding incoming webhook is working as expected.

**Pull request merged**